### ISSUE TITLE
How to handle the multispectral image (NOT RGB image)?

### ISSUE THREAD

**Issue by yifaliu** [qubvel/segmentation_models.pytorch#54]
I have multispectral image dataset (each image has eight channels) and want to feed them into the pretrained unet. In this case, how to modify the network so that the initialization of the first layer from the pretrained weight can be ignored. Otherwise, there will have error for the first layer's initialization due to the different input size of two models. Thanks.

**Comment by phillies** [comment]
You can just replace the first layer with a new Conv2d layer:

```
import torch
import segmentation_models_pytorch as smp

in_channels = 8

net = smp.Unet()
print(net.encoder.conv1)
# Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)

net.encoder.conv1 = torch.nn.Conv2d(in_channels=in_channels, out_channels=64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)

x = torch.randn(1,8,128,128) # test image
y = net(x)
```

**Comment by yifaliu** [comment]
Thanks. It works.

**Comment by bao18** [comment]
Hi, @phillies, thanks for the nice advice.
For prediction, Is any other modification.
The prediction "net.predict(X)" always gives 0 during training.

**Comment by phillies** [comment]
@bao18 This is strange, as `predict` does nothing else than calling the `forward` and `activation` methods but without changing the gradients. It's the same function as just calling `net(X)` and then applying the activation function (if any).

Check in your code if 
```
pred = net(X)
if net.activation:
    pred = net.activation(X)
```
also leads to 0 during training.

**Comment by qubvel** [comment]
add this feature in #100
